includes/class-wcsatt-order.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * WCS_ATT_Order class
+ *
+ * @author   SomewhereWarm <info@somewherewarm.gr>
+ * @package  WooCommerce Subscribe All the Things
+ * @since    2.0.0
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Order hooks for saving/restoring the subscription state of a product to/from order item data.
+ *
+ * @class    WCS_ATT_Order
+ * @version  2.0.0
+ */
+class WCS_ATT_Order {
+
+	public static function init() {
+
+		// Restore subscription data when creating a cart item using an order item as reference.
+		add_filter( 'woocommerce_order_again_cart_item_data', array( __CLASS__, 'restore_cart_item_from_order_item' ), 10, 3 );
```

Comment:
Should do the trick in combination with https://github.com/Prospress/woocommerce-subscribe-all-the-things/commit/bf7902962f722308eaee69ad548445423210870f